lib/graphql/schema/build_from_ast.rb

```diff
@@ -0,0 +1,260 @@
+              description: input_argument.description,
+              default_value: default_value,
+            )
+          ]
+        end
+      end
+
+      def build_directive(directive_definition, type_resolver)
+        GraphQL::Directive.define(
+          name: directive_definition.name,
+          description: directive_definition.description,
+          arguments: Hash[build_directive_arguments(directive_definition, type_resolver)],
+          locations: directive_definition.locations.map(&:to_sym),
+        )
+      end
+
+      def build_directive_arguments(directive_definition, type_resolver)
+        directive_definition.arguments.map do |directive_argument|
+          [
+            directive_argument.name,
+            GraphQL::Argument.define(
+              name: directive_argument.name,
+              type: type_resolver.call(directive_argument.type),
+              description: directive_argument.description,
+              default_value: directive_argument.default_value,
+            )
+          ]
+        end
+      end
+
+      def build_interface_type(interface_type_definition, type_resolver)
+        GraphQL::InterfaceType.define(
+          name: interface_type_definition.name,
+          description: interface_type_definition.description,
+          fields: Hash[build_fields(interface_type_definition.fields, type_resolver)],
+        )
+      end
+
+      def build_fields(field_definitions, type_resolver)
+        field_definitions.map do |field_definition|
+          field_arguments = Hash[field_definition.arguments.map do |argument|
+            default_value = case argument.default_value
+            when GraphQL::Language::Nodes::Enum
+              argument.default_value.name
+            else
+              argument.default_value
+            end
+
+            arg = GraphQL::Argument.define(
+              name: argument.name,
+              description: argument.description,
+              type: type_resolver.call(argument.type),
+              default_value: default_value,
+            )
+
+            [argument.name, arg]
+          end]
+
+          [
+            field_definition.name,
+            GraphQL::Field.define(
+              name: field_definition.name,
+              description: field_definition.description,
+              type: type_resolver.call(field_definition.type),
+              arguments: field_arguments,
+              deprecation_reason: build_deprecation_reason(field_definition.directives),
+            )
+          ]
+        end
+      end
+
+      def resolve_type(types, ast_node)
+        case ast_node
+        when GraphQL::Language::Nodes::TypeName
+          type_name = ast_node.name
+          case type_name
+          when "Int"
+            INT_TYPE
+          when "String"
+            STRING_TYPE
+          when "Float"
+            FLOAT_TYPE
+          when "Boolean"
+            BOOLEAN_TYPE
+          when "ID"
+            ID_TYPE
```

Comment:
Do you mean add all built-in types to the `types` hash above such that we can :fire: this extra logic? If so, yeah that's what I had in mind, but that's why I was wondering if we could have a `BUILT_IN_TYPES` constant hash somewhere so that we can do something like:

```ruby
types = {}
types.merge(GraphQL::Language::BUILT_IN_TYPES)
# ...
```

And :fire: this code.